googleapiclient/discovery_cache/file_cache.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2014 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""File based cache for the discovery document."""
```

Comment:
This makes me sad.

Potential for more pragmatic approach: just fail if the file doesn't exist and instruct users to run some command before to generate the file.

- Downside of this: more setup. Also if you wanted to cache by default it would be annoying to people who didn't realize what they needed to do
- Upside: less code (and in this case, code that is difficult to get right).

In context, Python is the only language (maybe JS in the browser but that's lame) where you don't need to actively download a generated library (or run a generator locally that builds code from a discovery doc). So I don't think it's that big a deal to ask user's to set up the file before running the code.

I should also mention there is a lot of atomic file saving guts in `oauth2client`. Maybe worth spinning it out into a library or something?